include/openrave/logging.h

```diff
@@ -200,7 +200,14 @@ OPENRAVE_API log4cxx::LevelPtr RaveGetVerboseLogLevel();
             } \
             /* vswprintf does not tell us how much space is needed, so we need to grow until it is satisfied */ \
             wslen *= 2; \
-            wsallocated = (wchar_t*)realloc(wsallocated, wslen*sizeof(wchar_t)); \
+            wchar_t* wsnext = (wchar_t*)realloc(wsallocated, wslen*sizeof(wchar_t)); \
+            /* realloc() can return NULL, need to remember previous allocation to avoid memory leaks */ \
+            if (wsnext != NULL) {  \
+                wsallocated = wsnext;  \
```

Comment:
I'm confused here. Don't you have to change `wslen` back to the original value? Otherwise `vswprintf` will happily write beyond the buffer, and now we have memory corruption.

Reply:
When we get a `NULL`, `wr` becomes `-1` and we get straight to this part (skipping printing):
```
        if (wsallocated != NULL) { \
            free(wsallocated); \
            wsallocated = NULL; \
        } \
```